bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -412,50 +427,78 @@ private void addIconColor(EList<ColorArray> iconColor, UIComponent component) {
 
     private void addColor(EList<ColorArray> color, UIComponent component, String key) {
         if (component.getConfig() != null && component.getConfig().containsKey(key)) {
-            for (Object sourceColor : (Collection<?>) component.getConfig().get(key)) {
-                if (sourceColor instanceof String) {
-                    Matcher matcher = COLOR_PATTERN.matcher(sourceColor.toString());
-                    if (matcher.matches()) {
+            Object sourceColors = component.getConfig().get(key);
+            if (sourceColors instanceof Collection<?>) {
+                for (Object sourceColor : (Collection<?>) sourceColors) {
+                    if (sourceColor instanceof String) {
+                        String argument = getRuleArgument(sourceColor.toString());
+                        List<String> conditionsString = getRuleConditions(sourceColor.toString(), argument);
                         ColorArrayImpl colorArray = (ColorArrayImpl) SitemapFactory.eINSTANCE.createColorArray();
-                        ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
-                        condition.setItem(matcher.group("item"));
-                        condition.setCondition(matcher.group("condition"));
-                        condition.setSign(matcher.group("sign"));
-                        condition.setState(matcher.group("state"));
-                        colorArray.eSet(SitemapPackage.COLOR_ARRAY__CONDITIONS, condition);
+                        colorArray.eSet(SitemapPackage.COLOR_ARRAY__ARG, argument);
```

Comment:
Using `colorArray.setArg` is probably proper.

Reply:
Done.